marcelo_zoccoler/entry_user_interf4/Readme.md

```diff
@@ -0,0 +1,335 @@
+            self.label_layer = self.viewer.add_labels(label)
+        else:
+            self.label_layer.data = label
+        self.horizontalSlider.setValue(level)
+        
+"""
+magicgui version
+"""
+from magicgui import magicgui
+from napari.types import ImageData, LabelsData
+@magic_factory(delta={'label': 'Temperature Increase (Δ°C):', 
+                                           'min': 0, 'max' : 3, 'step': 0.1},
+               new_level={'label':'Sea Level (dm):', 'widget_type':'Slider',
+                         'min': 0, 'max' : 255})
+def flood_magic_factory(image: ImageData, delta: float=0, new_level: int=0) -> LabelsData: 
+    new_level = delta*85
+    label_image = image <= new_level
+    label_image = label_image.astype(int)*13 # label 13 is blue in napari
+    return(label_image)
+
+"""
+FunctionGui version
+"""
+from magicgui.widgets import FunctionGui
+from napari.types import LayerDataTuple
+def flood_fgui(image: ImageData, delta: float=0, new_level: int=0) -> LayerDataTuple: 
+    new_level = delta*85
+    label_image = image <= new_level
+    label_image = label_image.astype(int)*13 # label 13 is blue in napari
+    return((label_image, {'name': 'flood result','metadata': {'new_level':new_level}}))
+
+class FunctionGui_flood(FunctionGui):
+    def __init__(self):
+        super().__init__(
+          flood_fgui,
+          call_button=True,
+          layout='vertical',
+          param_options={'delta':
+                             {'label': 'Temperature Increase (Δ°C):', 
+                              'min': 0, 'max' : 3, 'step': 0.1},
+                        'new_level':
+                            {'label':'Sea Level (dm):', 'widget_type':'Slider',
+                             'min': 0, 'max' : 255}}
+        )
+        
+    def __call__(self):
+        label_image = super().__call__()
+        new_level = round(label_image[1]['metadata']['new_level'])
+        self.new_level.value = new_level
+
+@napari_hook_implementation
+def napari_experimental_provide_dock_widget():
+    # you can return either a single widget, or a sequence of widgets
+    return [ExampleQWidget, example_magic_widget]
+```
+
+Last thing is to replace `napari_experimental_provide_dock_widget` outputs by our own. In our case, it should look like this:
+
+```Python
+@napari_hook_implementation
+def napari_experimental_provide_dock_widget():
+    # you can return either a single widget, or a sequence of widgets
+    return [Qt_Designer_flood, flood_magic_factory, FunctionGui_flood]
+```
+If you now call `napari` again, your plugin, along with each of the 3 sub-menu versions, should be available and working!
+
+![](images/flood_plugin_v1.png)
+
+Obs: remember we wrote our qt_designer version to work with a specific layer name (`napari_island`). There are other ways of selecting the appropriated layer, like by order, type, etc. You can have access to available layers through the viewer instance `viewer.layers`.
```

Comment:
What does "Obs" mean? Can you write it out? (-:

Reply:
ooops... I thought "Obs" was standard for an additional "observation". I think PS was the closest acronym I was looking for.
I will replace for 'Remider', I think it's best.